be/src/util/bitmap_value.h

```diff
@@ -1375,24 +1376,23 @@ class BitmapValue {
         case BITMAP:
             switch (_type) {
             case EMPTY:
-                _bitmap = rhs._bitmap;
+                _bitmap = std::make_shared<detail::Roaring64Map>(*rhs._bitmap);
                 _type = BITMAP;
                 break;
             case SINGLE:
-                if (rhs._bitmap->contains(_sv)) {
-                    rhs._bitmap->remove(_sv);
+                _bitmap = std::make_shared<detail::Roaring64Map>(*rhs._bitmap);
+                if (_bitmap->contains(_sv)) {
+                    _bitmap->remove(_sv);
                 } else {
-                    rhs._bitmap->add(_sv);
+                    _bitmap->add(_sv);
                 }
-                _bitmap = rhs._bitmap;
                 _type = BITMAP;
                 break;
             case BITMAP: {
-                BitmapValue lhs_bitmap(*this);
-                *_bitmap -= *rhs._bitmap;
-                *rhs._bitmap -= *lhs_bitmap._bitmap;
-
+                BitmapValue intersect_bitmap(*this);
                 *_bitmap |= *rhs._bitmap;
+                *intersect_bitmap._bitmap &= *rhs._bitmap;
```

Comment:
it that logically correct?

Reply:
a⊕b = (¬a ∧ b) ∨ (a ∧¬b) = (a V b) ∧ ¬(a ∧ b)
Before modification: (¬a ∧ b) ∨ (a ∧¬b)
after modification: (a V b) ∧ ¬(a ∧ b)

Reply:
> it that logically correct?

It's the same in terms of set operations

Reply:
a more efficient & straightforward way is:

```
*_bitmap -= *rhs._bitmap;
new X(*rhs._bitmap);
X -= *_bitmap;
*_bitmap |= X;

```

Reply:
No, my way is wrong.